### PR TITLE
implement blocking poll for mapping creation in ensure_index_mapping()

### LINKED ISSUE #220: Race condition exists when sweepers ensures index mappings and writes document updates very soon after
### Checked for duplicates

No - I haven't checked

### 🐛 Describe the bug

On MCP DEV (68 docs), repairkit's version metadata attribute was written but was unsearchable, causing infinite loop.

This is due to the mapping ensure function calling `put_mapping` and allowing execution to proceed while that update is still pending in AOSS.

### 🕵️ Expected behavior

I expected ensure_index_mapping to synchronously _ensure_ the presences of an index mapping.

### 📜 To Reproduce

dw

### 🖥 Environment Info

- Version of this software [e.g. vX.Y.Z]
- Operating System: [e.g. MacOSX with Docker Desktop vX.Y]
...

### 📚 Version of Software Used

_No response_

### 🩺 Test Data / Additional context

_No response_

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

_No response_

### 🎉 Integration & Test

_No response_

### PR DESCRIPTION
**Pull request by alexdunnjpl**
this addresses a potential race condition where writes occur very quickly after the put_mapping() call

## 🗒️ Summary
Addresses a race condition noticed by @tloubrieu-jpl on MCP-DEV.

### 🤖 AI Assistance Disclosure
Claude Web used for initial implementation, followed by excision/rework of 90% of it because it likes biting me any time I get comfortable enough to trust it with something simple, end-to-end.

## ⚙️ Test Data and/or Report
Existing test cases are sufficient

## ♻️ Related Issues
fixes #220 

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [x] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [x] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [x] **SonarCloud:** Confirmed no new High or Critical security findings.
- [x] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [x] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [x] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: src/pds/registrysweepers/utils/db/indexing.py [hunk 1]
```diff
@@ -1,10 +1,85 @@
...
+        UnexpectedMappingTypeError: immediately, if the mapping exists with a conflicting type.
+        MissingMappingError: if the mapping fails to propagate within the retry budget.
     """
-    client.indices.put_mapping(index=index_name, body={"properties": {property_name: {"type": property_type}}})
+    try:
+        _check_mapping(client, index_name, property_name, property_type)
+        return
+    except MissingMappingError:
+        logger.info(
+            f"Mapping for '{property_name}' not yet present in index '{index_name}' - creating mapping with type {property_type}.",
```
Comment (nutjob4life) [inline src/pds/registrysweepers/utils/db/indexing.py]: Extra `,` at the end of the line

File: src/pds/registrysweepers/utils/db/indexing.py [hunk 2]
```diff
@@ -1,10 +1,85 @@
...
+    )
+    return existing_properties.get(property_name, {}).get("type")
+
+
+def _check_mapping(client: OpenSearch, index_name: str, property_name: str, expected_property_type: str) -> None:
+    """
+    Checks whether the mapping for a given property has propagated with the expected type.
+    - Returns silently if the mapping is present and correct.
+    - Raises UnexpectedMappingTypeError if the type is wrong.
+    - Raises MissingMappingError if the mapping does not yet exist..
```
Comment (nutjob4life) [inline src/pds/registrysweepers/utils/db/indexing.py]: Extra `.` at end of the line

**Review by nutjob4life — APPROVED** [review]
Looks fine! I just had 2 microscopic nits to pick.

Approval: ✅

Even Tox is happy:
```
  py312: OK (7.91=setup[0.73]+cmd[7.18] seconds)
  py313: OK (5.42=setup[1.17]+cmd[4.26] seconds)
  docs: OK (1.60=setup[0.64]+cmd[0.96] seconds)
  lint: OK (4.83=setup[0.01]+cmd[4.82] seconds)
  congratulations :) (19.86 seconds)
```

File: src/pds/registrysweepers/utils/db/indexing.py (hunk 1 above)
Comment (alexdunnjpl) [reply src/pds/registrysweepers/utils/db/indexing.py]: _Some_ would consider superfluous commas

P   Y   T   H   O   N   I   C
Y
T
H
O
N
I
C

How else are you to know that the argument is just one element of a single-element sequence of arguments?!

Comment (nutjob4life) [reply src/pds/registrysweepers/utils/db/indexing.py]: @alexdunnjpl 😂 true, true!
